### PR TITLE
Automated UI testing using Galata

### PR DESCRIPTION
**Pull request by mbektas**
- Adds UI testing support using https://github.com/jupyterlab/galata
- Tests are run as part of CI with `ui-tests` workflow added
- After tests are run, a result report is generated and uploaded. It can be accessed in GitHub Actions page. If any of the tests fail, image diffs and failing call-stack can be inspected from report. Simply download the artifact and open `test/report/index.html` for the interactive report.
- Detailed documentation for adding new UI tests is added in `ui-tests/README.md`
- Helper npm scripts and a shell script added to help local development and testing. Usage described in the documentation as well.

## Test Plan
- [x] General UI Tests:
  - Launch screen, toggling Simple Mode, toggling Dark Theme
  - Open and capture main menu and sub menus
  - Open and capture sidebar tabs
  - Move sidebar panel to right
- [x] Notebook Create
  - Create cells for each cell type (raw, markdown, code)
  - Save and capture notebook
  - Toggle Dark Theme
- [x] Notebook Run
  - Upload notebook and image dependency to JupyterLab
  - Check File Browser list for uploaded files
  - Run notebook and capture notebook panel
  - Check numerical cell output values
- [x] Cell editing
  - Create a notebook with couple of cells
  - Modify notebook using operations in Edit menu (cut, copy, paste, delete, select, deselect, move, split, merge)
- [x] Notebook Toolbar
  - Create a notebook with couple of cells
  - Run actions in notebook toolbar
- [x] Table of Contents
  - Create tags
  - Create a notebook with markdown and code cells
  - Select cells and assign tags
  - Show TOC panel, filter using cell types and tags

Below is the capture from a test run:
jlab-galata

### CODE REVIEW

**Comment by mbektas** [comment]
here is a demo of detecting of visual regression caused by a CSS change: https://github.com/mbektas/jupyterlab/pull/6/checks?check_run_id=2743138752

and this is the image diff generated during that detection:
general_simple_mode

File: ui-tests/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "jupyterlab-ui-tests",
```
Comment (jtpio) [inline ui-tests/package.json:2]: Maybe we can follow the existing pattern and use the `@jupyterlab` scope for naming this package, even though it's private?

```suggestion
  "name": "@jupyterlab/ui-tests",
```

**Comment by jtpio** [comment]
For folks interested, you can find the test results uploaded as artifacts:

image

Then download the tarball, extract it, and open the `index.html` file to see the report:

image

**Review by jtpio — APPROVED** [review]
Thanks!

File: .github/workflows/ui-tests.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+    - name: Install dependencies
+      env:
+        GROUP: ${{ matrix.group }}
+      run: |
+        bash ./scripts/ci_install.sh
```
Comment (jtpio) [inline .github/workflows/ui-tests.yml:52]: At some point later, we could also extend `ci_install` to install specific dependencies for the `ui_tests` group, for example `xeus-python` to enable the debugger, similar to:

https://github.com/jupyterlab/jupyterlab/blob/9b5a6fd63df142424ec785e07993c8586cd7107c/scripts/ci_install.sh#L53-L56

**Review by blink1073 — APPROVED** [review]
Fantastic, thank you!

**Pull request merged**